test/plugins/buildClusters.test.js

```diff
@@ -0,0 +1,454 @@
+            });
+        });
+
+        it('returns 500 when the model encounters an error with an external cluster', () => {
+            const testError = new Error('datastoreSaveError');
+
+            buildClusterFactoryMock.create.rejects(testError);
+
+            return server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 500);
+            });
+        });
+
+        it('returns 500 when the model encounters an error with an internal cluster', () => {
+            const testError = new Error('datastoreSaveError');
+
+            options.payload = internalPayload;
+            params = internalParams;
+            buildClusterFactoryMock.create.rejects(testError);
+
+            return server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 500);
+            });
+        });
+    });
+
+    describe('DELETE /buildclusters/{name}', () => {
+        const name = 'iOS';
+        let options;
+        let buildClusterMock;
+        let userMock;
+
+        beforeEach(() => {
+            buildClusterMock = getMockBuildClusters(testBuildCluster);
+
+            options = {
+                method: 'DELETE',
+                url: `/buildclusters/${name}`,
+                credentials: {
+                    scope: ['user'],
+                    username,
+                    scmContext
+                }
+            };
+            userMock = {
+                username,
+                unsealToken: sinon.stub()
+            };
+            userFactoryMock.get.resolves(userMock);
+            buildClusterFactoryMock.list.resolves(buildClusterMock);
+            buildClusterMock.remove.resolves({});
+        });
+
+        it('returns 204 when delete is successful', () =>
+            server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 204);
+                assert.calledOnce(buildClusterMock.remove);
+            })
+        );
+
+        it('returns 403 when user does not have permission', () => {
+            screwdriverAdminDetailsMock.returns({ isAdmin: false });
+
+            return server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 403);
+            });
+        });
+
+        it('returns 404 when build cluster does not exist', () => {
+            buildClusterFactoryMock.list.resolves(null);
+
+            return server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 404);
+            });
+        });
+
+        it('returns 404 when user does not exist', () => {
+            userFactoryMock.get.withArgs({ username, scmContext }).resolves(null);
+
+            return server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 404);
+            });
+        });
+
+        it('returns 500 when call returns error', () => {
+            buildClusterFactoryMock.remove.rejects('collectionRemoveError');
```

Comment:
should be `buildClusterFactoryMock.remove.rejects(new Error('collectionRemoveError'));`